src/data/zmod/basic.lean

```diff
@@ -331,6 +331,38 @@ lemma cast_int_cast (k : ℤ) : ((k : zmod n) : R) = k :=
 
 end universal_property
 
+lemma int_coe_eq_int_coe_iff (a b : ℤ) (c : ℕ) :
+  (a : zmod c) = (b : zmod c) ↔ a ≡ b [ZMOD c] :=
+char_p.int_coe_eq_int_coe_iff (zmod c) c a b
+
+lemma nat_coe_eq_nat_coe_iff (a b c : ℕ) :
+  (a : zmod c) = (b : zmod c) ↔ a ≡ b [MOD c] :=
+begin
+  convert zmod.int_coe_eq_int_coe_iff a b c,
+  simp [nat.modeq.modeq_iff_dvd, int.modeq.modeq_iff_dvd],
+end
+
+@[simp]
+lemma int_coe_zmod_eq_zero_iff_dvd (a : ℤ) (b : ℕ) : (a : zmod b) = 0 ↔ (b : ℤ) ∣ a :=
```

Comment:
Not sure that this should be `@[simp]`. Equalities of `zmod` are often much more useful than `dvd`

Reply:
Sure; I've removed these `@[simp]`s.